src/oc_erchef/apps/oc_chef_wm/src/oc_chef_action_insights.erl

```diff
@@ -0,0 +1,67 @@
+%% Copyright 2012-2014 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+%% @doc Receiver of ingest requests. Requests are queued in a file and parsed
+%% asynchronously.
+
+-module(oc_chef_action_insights).
+
+-behaviour(gen_server).
+
+%% API
+-export([
+         ingest/2,
+         start_link/0
+        ]).
+
+%% gen_server callbacks
+-export([
+         init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3
+        ]).
+
+-spec ingest(ShouldSend :: boolean(), Data :: binary()) -> ok.
+ingest(true, Data) ->
+    gen_server:cast(?MODULE, {ingest, Data});
+ingest(false, _) -> ok.
+
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init([]) ->
+    file:open(envy:get(oc_chef_wm, insights_log, string), [append]).
```

Comment:
I think the reason this works is because init is expected to return {ok, State}. Since file:open returns {ok, IODevice} gen_server is okay with it. We could probably be more explicit about what is being returned, or as you suggested, return a record.